plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TableSnapshot.java

```diff
@@ -245,7 +250,73 @@ private CheckpointEntryIterator getCheckpointTransactionLogEntries(
         catch (IOException e) {
             throw new TrinoException(DELTA_LAKE_FILESYSTEM_ERROR, format("Unexpected IO exception occurred while retrieving the length of the file: %s for the table %s", checkpoint, table), e);
         }
-        return new CheckpointEntryIterator(
+        if (checkpoint.getV2Checkpoint().isPresent()) {
+            Optional<List<DeltaLakeTransactionLogEntry>> result;
```

Comment:
Can you encapsulate v2 logic in helper function so it's clearer what happens in case of v1?